### DISCUSSION TITLE
FSM for nested JSON?

### DISCUSSION THREAD

**Discussion by zifanwTF** [outlines-dev/outlines#732] [General]
Just reading the blog post and wonder if it is possible to have a FSM when the user wants the response to have a nested structure. Say the desired output is something like 

```
{"key": [dict1, dict2, dict3, ...]}
```

Each element of the list is a dictionary. In this case, when you are at the state "}", you cannot determine if the decoding is finished or not, right? As far as I know (pls correct me if I am wrong), the transition in a FSM only depends on the current stage but in this case you actually need to have a stack to count the if all "{" are closed? I haven't tried the package yet and just wonder if this is case is covered?

**Comment by rlouf** [comment]
You should try it out, but I think it will work.
